pkg/orgIdColumnPopulator/main.go

```diff
@@ -0,0 +1,178 @@
+type blessedLiteral string
+
+func MapAccountToOrgId(ctx context.Context, database *sql.DB, table, accountColumn, orgIdColumn string, nullOrgIdPlaceholder string, dbOperationTimeout int, batchSize int, batchTranslator tenantid.BatchTranslator, recordMetrics MetricsRecorder) error {
+
+	blessedTableName := blessedIdentifier(pq.QuoteIdentifier(table))
+	blessedAccountColumnName := blessedIdentifier(pq.QuoteIdentifier(accountColumn))
+	blessedOrgIdColumnName := blessedIdentifier(pq.QuoteIdentifier(orgIdColumn))
+	blessedNullOrgIdPlaceholder := blessedLiteral(pq.QuoteLiteral(nullOrgIdPlaceholder))
+
+	uniqueAccounts, rowsUpdated, mappingErr := mapAccountToOrgId(ctx, database, blessedTableName, blessedAccountColumnName, blessedOrgIdColumnName, blessedNullOrgIdPlaceholder, dbOperationTimeout, batchSize, batchTranslator)
+
+	// Record the metrics even if there was an error??
+	_ = recordMetrics(table, uniqueAccounts, rowsUpdated)
+
+	return mappingErr
+}
+
+func mapAccountToOrgId(ctx context.Context, database *sql.DB, table, accountColumn, orgIdColumn blessedIdentifier, nullOrgIdPlaceholder blessedLiteral, dbOperationTimeout int, batchSize int, batchTranslator tenantid.BatchTranslator) (int64, int64, error) {
+
+	log.Println("table:", table)
+	log.Println("accountColumn:", accountColumn)
+	log.Println("orgIdColumn:", orgIdColumn)
+	log.Println("batchSize:", batchSize)
+	log.Println("nullOrgIdPlaceholder: ", nullOrgIdPlaceholder)
+
+	var uniqueAccounts int64
+	var rowsUpdated int64
+
+	statement, err := buildAccountsQuery(database, table, accountColumn, orgIdColumn, nullOrgIdPlaceholder, batchSize)
+	if err != nil {
+		return uniqueAccounts, rowsUpdated, err
+	}
+	defer statement.Close()
+
+	for {
+
+		eans, err := readAccountsFromDatabase(ctx, database, statement, batchSize, dbOperationTimeout)
+		if err != nil {
+			return uniqueAccounts, rowsUpdated, err
+		}
+
+		log.Printf("Read %d unique accounts from table %s\n", len(eans), table)
+
+		if len(eans) == 0 {
+			return uniqueAccounts, rowsUpdated, nil
+		}
+
+		translationResults, err := batchTranslator.EANsToOrgIDs(ctx, eans)
+		if err != nil {
+			return uniqueAccounts, rowsUpdated, err
+		}
+
+		for _, results := range translationResults {
+
+			uniqueAccounts++
+
+			if results.Err != nil {
+				log.Printf("Skipping translation of EAN (%s) to OrgID (%s) due to error: %s\n", results.EAN, results.OrgID, results.Err)
+				continue
+			}
+
+			if results.EAN == nil {
+				log.Printf("Skipping nil EAN\n")
+				continue
+			}
+
+			updatedRowsCount, err := updateOrgIdInDatabase(ctx, database, table, accountColumn, orgIdColumn, nullOrgIdPlaceholder, *results.EAN, results.OrgID, dbOperationTimeout)
+			if err != nil {
+				return uniqueAccounts, rowsUpdated, err
+			}
+
+			if updatedRowsCount == 0 {
+				// I can't really think of why this would happen...but I really don't want
+				// this thing to just run forever...
+				log.Println("FAILED TO UPDATED ANY ROWS...Looks like we are done!!")
```

Comment:
I hit this locally, even when there are rows being updated. Is that expected?

Reply:
That's interesting.  Can you help me reproduce that?

Can you send me the output that it generated?

Reply:
Sure! This is what I'm getting. It successfully updates the one row that needs updating:
```
2022/04/06 14:19:53 Configured to discard metrics
2022/04/06 14:19:53 Using fake EAN translator impl
2022/04/06 14:19:53 table: "api_tenant"
2022/04/06 14:19:53 accountColumn: "account_id"
2022/04/06 14:19:53 orgIdColumn: "org_id"
2022/04/06 14:19:53 batchSize: 500
2022/04/06 14:19:53 nullOrgIdPlaceholder:  ''
2022/04/06 14:19:53 Read 2 unique accounts from table "api_tenant"
2022/04/06 14:19:53 FAILED TO UPDATED ANY ROWS...Looks like we are done!!
2022/04/06 14:19:53 METRICS:
2022/04/06 14:19:53 tableName:  api_tenant
2022/04/06 14:19:53 uniqueAccounts: 2
2022/04/06 14:19:53 rowsUpdated: 1
2022/04/06 14:19:53 Failed to update any rows.
```

Reply:
That is interesting.  Is results.RowsAffected() returning zero in the updateOrgIdInDatabase() method?  :exploding_head:   :)